kimchi/src/verifier_index.rs

```diff
@@ -146,6 +146,10 @@ where
 {
     /// Produces the [VerifierIndex] from the prover's [ProverIndex].
     pub fn verifier_index(&self) -> VerifierIndex<G> {
+        if let Some(verifier_index) = &self.verifier_index {
+            return verifier_index.clone();
+        }
```

Comment:
at the end of this function, cache the verifier_index into self.verifier_index before returning it

Reply:
We don't have a `&mut` reference.